tests/integration-tests/tests/storage/test_efs.py

```diff
@@ -68,6 +73,129 @@ def test_efs_same_az(region, scheduler, pcluster_config_reader, clusters_factory
+    """
+    _assert_subnet_az_relations(region, vpc_stack, expected_in_same_az=False)
+    mount_dir = "efs_mount_dir"
+    cluster_config = pcluster_config_reader(
+        mount_dir=mount_dir, efs_fs_id=existing_efs.cfn_resources["FileSystemResource"]
+    )
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+
+    mount_dir = "/" + mount_dir
+    # test file in efs exist
+    logging.info("Testing ebs {0} is correctly mounted".format(mount_dir))
+    result = remote_command_executor.run_remote_command("df | grep '{0}'".format(mount_dir))
+    assert_that(result.stdout).contains(mount_dir)
+
+    logging.info("cat hello_worlding")
+    file_name = existing_efs.random_file_name
+    remote_command_executor.run_remote_command(f"cat {mount_dir}/{file_name}")
+    scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+    _test_efs_correctly_mounted(remote_command_executor, mount_dir)
+    _test_efs_correctly_shared(remote_command_executor, mount_dir, scheduler_commands)
+    remote_command_executor.run_remote_command(f"cat {mount_dir}/{file_name}")
+
+
+@pytest.fixture()
+def existing_efs(vpc_stack, cfn_stacks_factory, request, region):
+    # efs stack contains a single efs resource
+    efs_template = Template()
+    efs_template.set_version("2010-09-09")
+    efs_template.set_description("EFS stack created for testing existing EFS")
+    efs_template.add_resource(FileSystem("FileSystemResource"))
+    efs_stack = CfnStack(
+        name="integ-tests-efs-{0}{1}{2}".format(
+            random_alphanumeric(),
+            "-" if request.config.getoption("stackname_suffix") else "",
+            request.config.getoption("stackname_suffix"),
+        ),
+        region=region,
+        template=efs_template.to_json(),
+    )
+    cfn_stacks_factory.create_stack(efs_stack)
+
+    # write file stack contains a mount target and a instance to write a empty file with random name into the efs
+    write_file_template = Template()
+    write_file_template.set_version("2010-09-09")
+    write_file_template.set_description("Stack to write a file to the existing EFS")
+    default_security_group_id = (
+        boto3.client("ec2", region_name=region)
+        .describe_security_groups(
+            Filters=[
+                {"Name": "vpc-id", "Values": [vpc_stack.cfn_outputs["VpcId"]]},
+                {"Name": "group-name", "Values": ["default"]},
+            ]
+        )
+        .get("SecurityGroups")[0]
+        .get("GroupId")
+    )
+    write_file_template.add_resource(
+        MountTarget(
+            "MountTargetResource",
+            FileSystemId=efs_stack.cfn_resources["FileSystemResource"],
+            SubnetId=vpc_stack.cfn_outputs["PublicSubnetId"],
+            SecurityGroups=[default_security_group_id],
+        )
+    )
+    efs_stack.random_file_name = random_alphanumeric()
```

Comment:
Is there any harm to call `random_alphanumeric()` multiple times?

Reply:
There is no harm and won't affect the test functionality using different random alphanumeric, but just for consistency. Moreover, If we can create all stacks in an integration stack as substacks, it would be better.

Reply:
In this fixture, I am deleting `write_file_stack` while keeping `efs_stack`. I am not sure if putting them in the same stack or substacks achieve the same effect.